### ISSUE TITLE
pip install cookiecutter without python-dev fails

### ISSUE THREAD

**Issue by SweetAndLow** [audreyr/cookiecutter#569]
OS: Ubuntu 14.04
Version: cookiecutter-1.2.1-py2.py3-none-any

### Problem
Simply installing the package doesn't work and required some extra effort to get it working.
It would be great if the error was easier to figure out or it handled the dependency better.

### Repro:
New system, install python-pip, pip install cookiecutter 

### Output: 
`Command /usr/bin/python -c "import setuptools, tokenize;__file__='/tmp/pip_build_root/ruamel.yaml/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /tmp/pip-zWVNPC-record/install-record.txt --single-version-externally-managed --compile failed with error code 1 in /tmp/pip_build_root/ruamel.yaml
Storing debug log for failure in /home/svetter/.pip/pip.log`

### Workaround
install python-dev on the ubuntu system and the errors is fixed.

**Comment by hackebrot** [comment]
Hi @SweetAndLow, is there any useful information in ``~/.pip/pip.log``? :worried:

**Comment by SweetAndLow** [comment]
Can't attach logs because you need write permissions to the repo for that. I'll attempt to past the important bits from .pip/pip.log. Both tracebacks look to be the same but are a little different so I'll add both. This doesn't look like an issues with cookiecutter and more like one with ruamel. 

```
copying ./loader.py -> build/lib.linux-x86_64-2.7/ruamel/yaml

copying ./LICENSE -> build/lib.linux-x86_64-2.7/ruamel/yaml

running build_ext

building '_ruamel_yaml' extension

creating build/temp.linux-x86_64-2.7

creating build/temp.linux-x86_64-2.7/ext

x86_64-linux-gnu-gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -I/usr/include/python2.7 -c ext/_ruamel_yaml.c -o build/temp.linux-x86_64-2.7/ext/_ruamel_yaml.o

ext/_ruamel_yaml.c:4:20: fatal error: Python.h: No such file or directory

 #include "Python.h"

                    ^

compilation terminated.

error: command 'x86_64-linux-gnu-gcc' failed with exit status 1

----------------------------------------
Cleaning up...
Command /usr/bin/python -c "import setuptools, tokenize;__file__='/tmp/pip_build_root/ruamel.yaml/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /tmp/pip-VegwXB-record/install-record.txt --single-version-externally-managed --compile failed with error code 1 in /tmp/pip_build_root/ruamel.yaml
Storing debug log for failure in /home/svetter/.pip/pip.log
```
and
```
copying ./LICENSE -> build/lib.linux-x86_64-2.7/ruamel/yaml
    running build_ext
    building '_ruamel_yaml' extension
    creating build/temp.linux-x86_64-2.7
    creating build/temp.linux-x86_64-2.7/ext
    x86_64-linux-gnu-gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -I/usr/include/python2.7 -c ext/_ruamel_yaml.c -o build/temp.linux-x86_64-2.7/ext/_ruamel_yaml.o
    ext/_ruamel_yaml.c:4:20: fatal error: Python.h: No such file or directory
     #include "Python.h"
                        ^
    compilation terminated.
    error: command 'x86_64-linux-gnu-gcc' failed with exit status 1
    Complete output from command /usr/bin/python -c "import setuptools, tokenize;__file__='/tmp/pip_build_root/ruamel.yaml/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /tmp/pip-VegwXB-record/install-record.txt --single-version-externally-managed --compile:
    author -> Anthon van der Neut

author_email -> a.van.der.neut@ruamel.eu

classifiers -> ['Development Status :: 4 - Beta', 'Intended Audience :: Developers', 'License :: OSI Approved :: MIT License', 'Operating System :: OS Independent', 'Programming Language :: Python', 'Programming Language :: Python :: 2.6', 'Programming Language :: Python :: 2.7', 'Programming Language :: Python :: 3.3', 'Programming Language :: Python :: 3.4', 'Programming Language :: Python :: 3.5', 'Programming Language :: Python :: Implementation :: CPython', 'Programming Language :: Python :: Implementation :: PyPy', 'Programming Language :: Python :: Implementation :: Jython', 'Topic :: Software Development :: Libraries :: Python Modules', 'Topic :: Text Processing :: Markup']

cmdclass -> {'install_lib': <class __main__.MyInstallLib at 0x7f5c01dd3ce8>}

description -> ruamel.yaml is a YAML parser/emitter that supports roundtrip preservation of comments, seq/map flow style, and map key order

entry_points -> None

ext_modules -> [<setuptools.extension.Extension instance at 0x7f5bff6fbe60>]

install_requires -> ['ruamel.base>=1.0.0', 'ruamel.ordereddict']

license -> MIT license

name -> ruamel.yaml

namespace_packages -> ['ruamel']

package_data -> {'ruamel.yaml': ['LICENSE']}

package_dir -> {'ruamel': '.ruamel', 'ruamel.yaml': '.'}

packages -> ['ruamel', 'ruamel.yaml']

url -> https://bitbucket.org/ruamel/yaml

version -> 0.10.12

```

**Comment by hackebrot** [comment]
Many thanks @SweetAndLow! :bow:

**Comment by bertdemiranda** [comment]
I am getting the same error when trying to install on OpenSUSE (LEAP 42.1):

pip install cookiecutter
Collecting cookiecutter
  Downloading cookiecutter-1.3.0-py2.py3-none-any.whl (46kB)
    100% |████████████████████████████████| 49kB 2.6MB/s 
Collecting whichcraft>=0.1.1 (from cookiecutter)
  Downloading whichcraft-0.1.1-py2.py3-none-any.whl
Collecting click>=5.0 (from cookiecutter)
  Downloading click-6.2-py2.py3-none-any.whl (70kB)
    100% |████████████████████████████████| 73kB 6.0MB/s 
Collecting jinja2>=2.7 (from cookiecutter)
  Downloading Jinja2-2.8-py2.py3-none-any.whl (263kB)
    100% |████████████████████████████████| 266kB 3.0MB/s 
Collecting future>=0.15.2 (from cookiecutter)
  Downloading future-0.15.2.tar.gz (1.6MB)
    100% |████████████████████████████████| 1.6MB 441kB/s 
Collecting binaryornot>=0.2.0 (from cookiecutter)
  Downloading binaryornot-0.4.0-py2.py3-none-any.whl
Collecting ruamel.yaml>=0.10.12 (from cookiecutter)
  Downloading ruamel.yaml-0.10.12.tar.gz (234kB)
    100% |████████████████████████████████| 237kB 2.1MB/s 
Collecting MarkupSafe (from jinja2>=2.7->cookiecutter)
  Downloading MarkupSafe-0.23.tar.gz
Collecting chardet>=2.0.0 (from binaryornot>=0.2.0->cookiecutter)
  Downloading chardet-2.3.0.tar.gz (164kB)
    100% |████████████████████████████████| 167kB 3.0MB/s 
Collecting ruamel.base>=1.0.0 (from ruamel.yaml>=0.10.12->cookiecutter)
  Downloading ruamel.base-1.0.0-py3-none-any.whl
Building wheels for collected packages: future, ruamel.yaml, MarkupSafe, chardet
  Running setup.py bdist_wheel for future
  Stored in directory: /home/bert/.cache/pip/wheels/d9/04/36/6bd807b5148e7c929d8f0991cc943a81f3287030a1b352e3fc
  Running setup.py bdist_wheel for ruamel.yaml
  Complete output from command /home/bert/python/.envs/desktop7/bin/python3 -c "import setuptools;__file__='/tmp/pip-build-78g36s12/ruamel.yaml/setup.py';exec(compile(open(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" bdist_wheel -d /tmp/tmpjeggpt3spip-wheel-:
  author -> Anthon van der Neut
  author_email -> a.van.der.neut@ruamel.eu
  classifiers -> ['Development Status :: 4 - Beta', 'Intended Audience :: Developers', 'License :: OSI Approved :: MIT License', 'Operating System :: OS Independent', 'Programming Language :: Python', 'Programming Language :: Python :: 2.6', 'Programming Language :: Python :: 2.7', 'Programming Language :: Python :: 3.3', 'Programming Language :: Python :: 3.4', 'Programming Language :: Python :: 3.5', 'Programming Language :: Python :: Implementation :: CPython', 'Programming Language :: Python :: Implementation :: PyPy', 'Programming Language :: Python :: Implementation :: Jython', 'Topic :: Software Development :: Libraries :: Python Modules', 'Topic :: Text Processing :: Markup']
  cmdclass -> {'install_lib': <class '__main__.MyInstallLib'>}
  description -> ruamel.yaml is a YAML parser/emitter that supports roundtrip preservation of comments, seq/map flow style, and map key order
  entry_points -> None
  ext_modules -> [<setuptools.extension.Extension object at 0x7fe097e72a58>]
  install_requires -> ['ruamel.base>=1.0.0']
  license -> MIT license
  name -> ruamel.yaml
  namespace_packages -> ['ruamel']
  package_data -> {'ruamel.yaml': ['LICENSE']}
  package_dir -> {'ruamel': '.ruamel', 'ruamel.yaml': '.'}
  packages -> ['ruamel', 'ruamel.yaml']
  url -> https://bitbucket.org/ruamel/yaml
  version -> 0.10.12
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.linux-x86_64-3.4
  creating build/lib.linux-x86_64-3.4/ruamel
  copying .ruamel/__init__.py -> build/lib.linux-x86_64-3.4/ruamel
  creating build/lib.linux-x86_64-3.4/ruamel/yaml
  copying ./scalarstring.py -> build/lib.linux-x86_64-3.4/ruamel/yaml
  copying ./constructor.py -> build/lib.linux-x86_64-3.4/ruamel/yaml
  copying ./compat.py -> build/lib.linux-x86_64-3.4/ruamel/yaml
  copying ./emitter.py -> build/lib.linux-x86_64-3.4/ruamel/yaml
  copying ./composer.py -> build/lib.linux-x86_64-3.4/ruamel/yaml
  copying ./events.py -> build/lib.linux-x86_64-3.4/ruamel/yaml
  copying ./setup.py -> build/lib.linux-x86_64-3.4/ruamel/yaml
  copying ./representer.py -> build/lib.linux-x86_64-3.4/ruamel/yaml
  copying ./__init__.py -> build/lib.linux-x86_64-3.4/ruamel/yaml
  copying ./cyaml.py -> build/lib.linux-x86_64-3.4/ruamel/yaml
  copying ./reader.py -> build/lib.linux-x86_64-3.4/ruamel/yaml
  copying ./error.py -> build/lib.linux-x86_64-3.4/ruamel/yaml
  copying ./configobjwalker.py -> build/lib.linux-x86_64-3.4/ruamel/yaml
  copying ./dumper.py -> build/lib.linux-x86_64-3.4/ruamel/yaml
  copying ./nodes.py -> build/lib.linux-x86_64-3.4/ruamel/yaml
  copying ./resolver.py -> build/lib.linux-x86_64-3.4/ruamel/yaml
  copying ./serializer.py -> build/lib.linux-x86_64-3.4/ruamel/yaml
  copying ./loader.py -> build/lib.linux-x86_64-3.4/ruamel/yaml
  copying ./comments.py -> build/lib.linux-x86_64-3.4/ruamel/yaml
  copying ./scanner.py -> build/lib.linux-x86_64-3.4/ruamel/yaml
  copying ./parser_.py -> build/lib.linux-x86_64-3.4/ruamel/yaml
  copying ./tokens.py -> build/lib.linux-x86_64-3.4/ruamel/yaml
  copying ./make_win_whl.py -> build/lib.linux-x86_64-3.4/ruamel/yaml
  copying ./main.py -> build/lib.linux-x86_64-3.4/ruamel/yaml
  copying ./LICENSE -> build/lib.linux-x86_64-3.4/ruamel/yaml
  running build_ext
  building '_ruamel_yaml' extension
  creating build/temp.linux-x86_64-3.4
  creating build/temp.linux-x86_64-3.4/ext
  gcc -pthread -Wno-unused-result -Werror=declaration-after-statement -DNDEBUG -fmessage-length=0 -grecord-gcc-switches -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector -funwind-tables -fasynchronous-unwind-tables -g -DOPENSSL_LOAD_CONF -fPIC -I/usr/include/python3.4m -c ext/_ruamel_yaml.c -o build/temp.linux-x86_64-3.4/ext/_ruamel_yaml.o
  ext/_ruamel_yaml.c:4:20: fatal error: Python.h: No such file or directory
   #include "Python.h"
                      ^
  compilation terminated.
  error: command 'gcc' failed with exit status 1
  
  ----------------------------------------
  Failed building wheel for ruamel.yaml
  Running setup.py bdist_wheel for MarkupSafe
  Stored in directory: /home/bert/.cache/pip/wheels/94/a7/79/f79a998b64c1281cb99fa9bbd33cfc9b8b5775f438218d17a7
  Running setup.py bdist_wheel for chardet
  Stored in directory: /home/bert/.cache/pip/wheels/a4/24/b6/124a2516812185506f0573d4abbeb7d08596f53618f8cc960d
Successfully built future MarkupSafe chardet
Failed to build ruamel.yaml
Installing collected packages: whichcraft, click, MarkupSafe, jinja2, future, chardet, binaryornot, ruamel.base, ruamel.yaml, cookiecutter
  Running setup.py install for ruamel.yaml
    Complete output from command /home/bert/python/.envs/desktop7/bin/python3 -c "import setuptools, tokenize;__file__='/tmp/pip-build-78g36s12/ruamel.yaml/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /tmp/pip-ozuqy3pj-record/install-record.txt --single-version-externally-managed --compile --install-headers /home/bert/python/.envs/desktop7/include/site/python3.4/ruamel.yaml:
    author -> Anthon van der Neut
    author_email -> a.van.der.neut@ruamel.eu
    classifiers -> ['Development Status :: 4 - Beta', 'Intended Audience :: Developers', 'License :: OSI Approved :: MIT License', 'Operating System :: OS Independent', 'Programming Language :: Python', 'Programming Language :: Python :: 2.6', 'Programming Language :: Python :: 2.7', 'Programming Language :: Python :: 3.3', 'Programming Language :: Python :: 3.4', 'Programming Language :: Python :: 3.5', 'Programming Language :: Python :: Implementation :: CPython', 'Programming Language :: Python :: Implementation :: PyPy', 'Programming Language :: Python :: Implementation :: Jython', 'Topic :: Software Development :: Libraries :: Python Modules', 'Topic :: Text Processing :: Markup']
    cmdclass -> {'install_lib': <class '__main__.MyInstallLib'>}
    description -> ruamel.yaml is a YAML parser/emitter that supports roundtrip preservation of comments, seq/map flow style, and map key order
    entry_points -> None
    ext_modules -> [<setuptools.extension.Extension object at 0x7f3dac5412e8>]
    install_requires -> ['ruamel.base>=1.0.0']
    license -> MIT license
    name -> ruamel.yaml
    namespace_packages -> ['ruamel']
    package_data -> {'ruamel.yaml': ['LICENSE']}
    package_dir -> {'ruamel.yaml': '.', 'ruamel': '.ruamel'}
    packages -> ['ruamel', 'ruamel.yaml']
    url -> https://bitbucket.org/ruamel/yaml
    version -> 0.10.12
    running install
    running build
    running build_py
    running build_ext
    building '_ruamel_yaml' extension
    gcc -pthread -Wno-unused-result -Werror=declaration-after-statement -DNDEBUG -fmessage-length=0 -grecord-gcc-switches -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector -funwind-tables -fasynchronous-unwind-tables -g -DOPENSSL_LOAD_CONF -fPIC -I/usr/include/python3.4m -c ext/_ruamel_yaml.c -o build/temp.linux-x86_64-3.4/ext/_ruamel_yaml.o
    ext/_ruamel_yaml.c:4:20: fatal error: Python.h: No such file or directory
     #include "Python.h"
                        ^
    compilation terminated.
    error: command 'gcc' failed with exit status 1
    
    ----------------------------------------
Command "/home/bert/python/.envs/desktop7/bin/python3 -c "import setuptools, tokenize;__file__='/tmp/pip-build-78g36s12/ruamel.yaml/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /tmp/pip-ozuqy3pj-record/install-record.txt --single-version-externally-managed --compile --install-headers /home/bert/python/.envs/desktop7/include/site/python3.4/ruamel.yaml" failed with error code 1 in /tmp/pip-build-78g36s12/ruamel.yaml   

Then I installed the python-devel package with YaST (could not locate a package python-dev),
but this does not make a difference.

I do have a virtual environment active, but do not see a ~/.pip/pip.log file; is there anywhere else I can look?

Can you advise a solution or workaround?

Thanks!
Bert.

**Comment by storax** [comment]
Is this related to missing dependencies of ruamel yaml?
Maybe this helps on Debian systems: `sudo apt-get install libyaml-dev python-dev python3-dev`
or on CentOS: `sudo yum install libyaml-devel python-devel`

**Comment by pydanny** [comment]
As Cookiecutter's use of YAML is limited, would it make sense to get rid of the external YAML parser and create our own simple one?

**Comment by pydanny** [comment]
Or perhaps find a really simple pure Python YAML parser as a third party package?

**Comment by hackebrot** [comment]
Yes, I think we should get rid of it altogether. :worried: 

Do you think ppl use advanced yaml features?

**Comment by bertdemiranda** [comment]
from storax:
>Is this related to missing dependencies of ruamel yaml?
>Maybe this helps on Debian systems: sudo apt-get install libyaml-dev python-dev python3-dev
>or on CentOS: sudo yum install libyaml-devel python-devel

I installed python3-devel using YaST, and could subsequently install cookiecutter
via 
pip install cookiecutter

and it works!

Thanks for your help and I wish you a great 2016!

Bert.

**Comment by pydanny** [comment]
Removing YAML from the project will fix a lot of problems. See #743

**Comment by hackebrot** [comment]
This issue has been resolved in version 1.4.0 by switching to [poyo](https://github.com/hackebrot/poyo).
